src/System.Memory/tests/ReadOnlyMemory/Equality.cs

```diff
@@ -84,6 +84,58 @@ public static void EmptyMemoryNotUnified()
             Assert.True(right.Equals(left));
         }
 
+        [Fact]
+        public static void EqualityThroughInterface_True()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<ReadOnlyMemory<int>> left = new ReadOnlyMemory<int>(array, 2, 3);
+            IEquatable<ReadOnlyMemory<int>> right = new ReadOnlyMemory<int>(array, 2, 3);
+
+            Assert.True(left.Equals(right));
+            Assert.True(right.Equals(left));
+        }
+
+        [Fact]
+        public static void EqualityThroughInterface_Reflexivity()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
```

Comment:
While we are here, can we add some tests for T != int as well?

For instance T == char (from string), in here: https://github.com/dotnet/corefx/blob/d3eed78a3765aef0325384c61a4593570ae3622b/src/System.Memory/tests/ReadOnlyMemory/Strings.cs#L161

And T == string.

Reply:
Would I do the same as the tests for `int`'s, but instead with `char` and `string`s?

Reply:
+ Do you mean that we add tests that compare two Memory<string> (or Memory<char>) using IEquatable interfaces?

Reply:
Added the cases for `string` and `char`.